File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/CoalProductSpecifications.cpp
// CoalProductSpecifications.cpp 
#include "CoalProductSpecifications.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CoalProductSpecifications::CoalProductSpecifications(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //standardQualityNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* standardQualityNode = xmlNode->FirstChildElement("standardQuality");

   if(standardQualityNode){standardQualityIsNull_ = false;}
   else{standardQualityIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- standardQualityNode , address : " << standardQualityNode << std::endl;
   #endif
   if(standardQualityNode)
   {
      if(standardQualityNode->Attribute("href") || standardQualityNode->Attribute("id"))
      {
          if(standardQualityNode->Attribute("id"))
          {
             standardQualityIDRef_ = standardQualityNode->Attribute("id");
             standardQuality_ = boost::shared_ptr<CoalStandardQuality>(new CoalStandardQuality(standardQualityNode));
             standardQuality_->setID(standardQualityIDRef_);
             IDManager::instance().setID(standardQualityIDRef_,standardQuality_);
          }
          else if(standardQualityNode->Attribute("href")) { standardQualityIDRef_ = standardQualityNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { standardQuality_ = boost::shared_ptr<CoalStandardQuality>(new CoalStandardQuality(standardQualityNode));}
   }

   //standardQualityScheduleNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* standardQualityScheduleNode = xmlNode->FirstChildElement("standardQualitySchedule");

   if(standardQualityScheduleNode){standardQualityScheduleIsNull_ = false;}
   else{standardQualityScheduleIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- standardQualityScheduleNode , address : " << standardQualityScheduleNode << std::endl;
   #endif
   if(standardQualityScheduleNode)
   {
      if(standardQualityScheduleNode->Attribute("href") || standardQualityScheduleNode->Attribute("id"))
      {
          if(standardQualityScheduleNode->Attribute("id"))
          {
             standardQualityScheduleIDRef_ = standardQualityScheduleNode->Attribute("id");
             standardQualitySchedule_ = boost::shared_ptr<CoalStandardQualitySchedule>(new CoalStandardQualitySchedule(standardQualityScheduleNode));
             standardQualitySchedule_->setID(standardQualityScheduleIDRef_);
             IDManager::instance().setID(standardQualityScheduleIDRef_,standardQualitySchedule_);
          }
          else if(standardQualityScheduleNode->Attribute("href")) { standardQualityScheduleIDRef_ = standardQualityScheduleNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { standardQualitySchedule_ = boost::shared_ptr<CoalStandardQualitySchedule>(new CoalStandardQualitySchedule(standardQualityScheduleNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CoalStandardQuality> CoalProductSpecifications::getStandardQuality()
{
   if(!this->standardQualityIsNull_){
        if(standardQualityIDRef_ != ""){
             return boost::shared_static_cast<CoalStandardQuality>(IDManager::instance().getID(standardQualityIDRef_));
        }else{
             return this->standardQuality_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CoalStandardQuality>();
   }
}
boost::shared_ptr<CoalStandardQualitySchedule> CoalProductSpecifications::getStandardQualitySchedule()
{
   if(!this->standardQualityScheduleIsNull_){
        if(standardQualityScheduleIDRef_ != ""){
             return boost::shared_static_cast<CoalStandardQualitySchedule>(IDManager::instance().getID(standardQualityScheduleIDRef_));
        }else{
             return this->standardQualitySchedule_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CoalStandardQualitySchedule>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/CoalProductSpecifications.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CoalProductSpecifications : ISerialized
    {
        public CoalProductSpecifications(XmlNode xmlNode)
        {
            XmlNode standardQualityNode = xmlNode.SelectSingleNode("standardQuality");
            
            if (standardQualityNode != null)
            {
                if (standardQualityNode.Attributes["href"] != null || standardQualityNode.Attributes["id"] != null) 
                {
                    if (standardQualityNode.Attributes["id"] != null) 
                    {
                        standardQualityIDRef_ = standardQualityNode.Attributes["id"].Value;
                        CoalStandardQuality ob = new CoalStandardQuality(standardQualityNode);
                        IDManager.SetID(standardQualityIDRef_, ob);
                    }
                    else if (standardQualityNode.Attributes["href"] != null)
                    {
                        standardQualityIDRef_ = standardQualityNode.Attributes["href"].Value;
                    }
                    else
                    {
                        standardQuality_ = new CoalStandardQuality(standardQualityNode);
                    }
                }
                else
                {
                    standardQuality_ = new CoalStandardQuality(standardQualityNode);
                }
            }
            
        
            XmlNode standardQualityScheduleNode = xmlNode.SelectSingleNode("standardQualitySchedule");
            
            if (standardQualityScheduleNode != null)
            {
                if (standardQualityScheduleNode.Attributes["href"] != null || standardQualityScheduleNode.Attributes["id"] != null) 
                {
                    if (standardQualityScheduleNode.Attributes["id"] != null) 
                    {
                        standardQualityScheduleIDRef_ = standardQualityScheduleNode.Attributes["id"].Value;
                        CoalStandardQualitySchedule ob = new CoalStandardQualitySchedule(standardQualityScheduleNode);
                        IDManager.SetID(standardQualityScheduleIDRef_, ob);
                    }
                    else if (standardQualityScheduleNode.Attributes["href"] != null)
                    {
                        standardQualityScheduleIDRef_ = standardQualityScheduleNode.Attributes["href"].Value;
                    }
                    else
                    {
                        standardQualitySchedule_ = new CoalStandardQualitySchedule(standardQualityScheduleNode);
                    }
                }
                else
                {
                    standardQualitySchedule_ = new CoalStandardQualitySchedule(standardQualityScheduleNode);
                }
            }
            
        
        }
        
    
        #region StandardQuality_
        private CoalStandardQuality standardQuality_;
        public CoalStandardQuality StandardQuality_
        {
            get
            {
                if (this.standardQuality_ != null)
                {
                    return this.standardQuality_; 
                }
                else if (this.standardQualityIDRef_ != null)
                {
                    standardQuality_ = IDManager.getID(standardQualityIDRef_) as CoalStandardQuality;
                    return this.standardQuality_; 
                }
                else
                {
                      return this.standardQuality_; 
                }
            }
            set
            {
                if (this.standardQuality_ != value)
                {
                    this.standardQuality_ = value;
                }
            }
        }
        #endregion
        
        public string standardQualityIDRef_ { get; set; }
        #region StandardQualitySchedule_
        private CoalStandardQualitySchedule standardQualitySchedule_;
        public CoalStandardQualitySchedule StandardQualitySchedule_
        {
            get
            {
                if (this.standardQualitySchedule_ != null)
                {
                    return this.standardQualitySchedule_; 
                }
                else if (this.standardQualityScheduleIDRef_ != null)
                {
                    standardQualitySchedule_ = IDManager.getID(standardQualityScheduleIDRef_) as CoalStandardQualitySchedule;
                    return this.standardQualitySchedule_; 
                }
                else
                {
                      return this.standardQualitySchedule_; 
                }
            }
            set
            {
                if (this.standardQualitySchedule_ != value)
                {
                    this.standardQualitySchedule_ = value;
                }
            }
        }
        #endregion
        
        public string standardQualityScheduleIDRef_ { get; set; }
        public string choiceStr_0;
        
    
        
    
    }
    
}

